src/plugins/discover/public/application/angular/doc_table/components/table_row.tsx

```diff
@@ -0,0 +1,258 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { Fragment, useCallback, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiIcon,
+  EuiLink,
+  EuiTitle,
+  EuiFlexItem,
+} from '@elastic/eui';
+import { getServices, IndexPattern } from '../../../../kibana_services';
+import { DOC_HIDE_TIME_COLUMN_SETTING } from '../../../../../common';
+import { TableCell } from './table_row/table_cell';
+import { formatRow, formatTopLevelObject } from '../../helpers';
+import { getContextUrl } from '../../../helpers/get_context_url';
+import { DocViewer } from '../../../components/doc_viewer/doc_viewer';
+import { DocViewFilterFn, ElasticSearchHit } from '../../../doc_views/doc_views_types';
+
+// guesstimate at the minimum number of chars wide cells in the table should be
+const MIN_LINE_LENGTH = 20;
+
+export type DocTableRow = ElasticSearchHit & {
+  isAnchor?: boolean;
+};
+
+export interface TableRowProps {
+  columns: string[];
+  filter: DocViewFilterFn;
+  indexPattern: IndexPattern;
+  row: DocTableRow;
+  onAddColumn?: (column: string) => void;
+  onRemoveColumn?: (column: string) => void;
+  useNewFieldsApi: boolean;
+}
+
+export const TableRow = ({
+  columns,
+  filter,
+  row,
+  indexPattern,
+  useNewFieldsApi,
+  onAddColumn,
+  onRemoveColumn,
+}: TableRowProps) => {
+  const services = getServices();
+
+  const [open, setOpen] = useState(false);
+  const docTableRowClassName = classNames('kbnDocTable__row', {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    'kbnDocTable__row--highlight': row.isAnchor,
+  });
+  const anchorDocTableRowSubj = row.isAnchor ? ' docTableAnchorRow' : '';
+
+  const flattenedRow = useMemo(() => indexPattern.flattenHit(row), [indexPattern, row]);
+  const mapping = useMemo(() => indexPattern.fields.getByName, [indexPattern]);
+  const hideTimeColumn = useMemo(
+    () => services.uiSettings.get(DOC_HIDE_TIME_COLUMN_SETTING, false),
+    [services]
+  );
+
+  // toggle display of the rows details, a full list of the fields from each row
+  const toggleRow = () => setOpen((prevOpen) => !prevOpen);
+
+  /**
+   * Fill an element with the value of a field
+   */
+  const _displayField = useCallback(
+    (fieldName: string, truncate = false) => {
+      const text = indexPattern.formatField(row, fieldName);
+      // eslint-disable-next-line react/no-danger
+      const fieldElement = <span dangerouslySetInnerHTML={{ __html: text }} />;
+
+      if (truncate && text.length > MIN_LINE_LENGTH) {
+        return <div className="truncate-by-height">{fieldElement}</div>;
+      }
```

Comment:
This `if` should not be needed anymore. The `text` as returned from `formatField` will always be wrapped in a `<span ng-non-bindable=""></span>`, thus even the shortest value will be 32 characters long and above the threshold here :D

You can also see this happening on master already when having a short field, that's nevertheless having a `truncate-by-height`:

![screenshot-20210715-113603](https://user-images.githubusercontent.com/877229/125766143-ff528478-7469-4f23-b488-761e321762cd.png)

I also think that's the "right" behavior that we want, since we always want to make sure those lines are truncated by the configured height, and not just if the value is longer than that. So I think we can remove this `if` and instead always render the `truncate-by-height` div around the value.

Reply:
That said, I think we also need once we're done with this another PR to clean up a bit of the truncation behavior in general, but I think this is a minimal change we can already make as part of this PR, since that code does not have any effect today.

Reply:
Moreover, I found that we have `ng-non-bindable` directive, which is also part of angular. For now we are removing it by `trimAngularSpan` util function, but it should be adjusted in index pattern formatter method in the follow up PRs, right?